vispy/visuals/filters/mesh.py

```diff
@@ -347,29 +349,38 @@ def _detach(self, visual):
     vec3 d = fwidth(v_bc);
     vec3 a3 = smoothstep(vec3(0.0), $width * d, v_bc);
     float factor = min(min(a3.x, a3.y), a3.z);
-    gl_FragColor = mix($color, gl_FragColor, factor);
-//    gl_FragColor = vec4(0.0, 0.0, 0.0, 0.95 * (1.0 - factor));
+    if ($enabled == 1) {
```

Comment:
We could exit early, at the top of the function, to avoid unnecessary computations:
```glsl
if ($enabled != 1) {
    return;
}
```

Reply:
:man_facepalming: Of course. Done.